### PR TITLE
Cleaned up the code and fix the logger for new required type by GCP API

### PR DESCRIPTION
**Pull request by jlauzy**
## Summary:
- The logger no longer works because GCP's API made a breaking change where the resource type is required.
- Cleaned up the code to have more flexibility for these changes by exposing more configuration to the client.

image

I'm pretty sure they are going to make a change to their SDK so that this is no longer nullable but required
image

Issue: XXX-XXXX

## Test plan:
Works Again!
image

### CODE REVIEW

File: packages/wonder-stuff-ci/src/get-mobile-release-logger.ts [hunk 1]
```diff
@@ -7,50 +7,36 @@ import {
...
-        redirectToStdout: boolean;
-        labels: {[key: string]: string};
-        logLevel: GCPLogLevels;
-        logName: string;
-        projectId: string;
-    } = {
-        redirectToStdout: false,
-        labels: {},
-        logLevel: "info",
-        logName: "release-raccoon",
```
Comment (jlauzy) [inline packages/wonder-stuff-ci/src/get-mobile-release-logger.ts:39]: This looked very gnarly for a function signature, so I turned it into a type to make it more readable.

File: packages/wonder-stuff-ci/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
         "node": ">=16"
     },
     "name": "@khanacademy/wonder-stuff-ci",
-    "version": "1.1.0",
+    "version": "1.0.0",
```
Comment (jeresig) [inline packages/wonder-stuff-ci/package.json]: Err, that doesn't seem right! I think you have to revert this change - the version changes will come as part of the Changesets release process.
